dev/io.openliberty.checkpoint_fat_mail/fat/src/io/openliberty/checkpoint/fat/IMAPTest.java

```diff
@@ -0,0 +1,219 @@
+    // BeforeClass method
+    private static GreenMail imapServer = null;
+
+    @ClassRule
+    public static RepeatTests r = RepeatTests.withoutModification()
+                    .andWith(new JakartaEE9Action().forServers(SERVER_NAME).fullFATOnly())
+                    .andWith(new JakartaEE10Action().forServers(SERVER_NAME).fullFATOnly());
+
+    /**
+     * TestDescription:
+     * This test is test the ability for the javaMail-1.6 feature to have a javax.mail.Session
+     * object defined on the IMAPMailSessionServlet. This Tests the features ability to define a mailSession
+     * object in the server.xml. The mailSession object is created by the MailSessionService class
+     * and injected into a Session object on the IMAPMailSessionServlet using a JNDI lookup. Using the
+     * IMAP protocol it access the inbox folder on the GreenMail test server. If the folder is accessed
+     * than it prints out a message if an exception isn't thrown, and that message is used in the
+     * assert in validate that the message was printed to the log
+     *
+     * This test is for FatSuiteLite
+     */
+    @Test
+    public void testMailSessionMail() throws Exception {
+
+        URL url = new URL("http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/IMAPTestingApp/IMAPMailSessionServlet");
+        Log.info(c, "testAutoInstall",
+                 "Calling MailSession Application with URL=" + url.toString());
+        HttpURLConnection con = getHttpConnection(url);
+        BufferedReader br = getConnectionStream(con);
+        String line = br.readLine();
+
+        // Check if IMAP protocol Session was able to access Inbox and check MimeMessage
+        assertNotNull("FAIL: IMAP folder was unable to access message.",
+                      server.waitForStringInLog(SENT));
+    }
+
+    private static void setupApp() throws Exception {
+        WebArchive testingApp = ShrinkWrap.create(WebArchive.class, "IMAPTestingApp.war")
+                        .addPackages(true, "IMAPTestingApp")
+                        .addAsWebInfResource(new File("test-applications/IMAPTestingApp/resources/META-INF/MANIFEST.MF"))
+                        .addAsWebInfResource(new File("test-applications/IMAPTestingApp/resources/META-INF/permissions.xml"));
+        ShrinkHelper.exportAppToServer(server, testingApp);
+
+    }
+
+    /**
+     * The BeforeClass method sets up the GreenMail test server, so that the "server" is waiting
+     * before even the Liberty class is set up.
+     *
+     * @throws UserException
+     * @throws Exception
+     */
+    @BeforeClass
+    public static void startGreenMail() throws Exception {
+        setupApp();
+        if (server.isStarted() != true) {
+            server.setCheckpoint(CheckpointPhase.AFTER_APP_START, false, null);
+            server.startServer();
+            // Update the mailSession config with correct password
+            updateVariableConfig(server, "imap_password", "imapPa$$word");
```

Comment:
This will change:
`<variable name="imap_password" defaultValue="wrong_password" />`
to be:
`<variable name="imap_password" value="wrong_pa$$word" />`
in the `server.xml`

Can you instead, leave the `server.xml` unchanged for restore and set the `server.env` var `IMAP_PASSWORD=wrong_pa$$word`

Reply:
used server.env to use the updated password.